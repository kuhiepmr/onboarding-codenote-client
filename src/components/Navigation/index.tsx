import { Auth } from 'aws-amplify';
import React, { Fragment } from 'react';
import { Nav, Navbar, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import { Link } from 'react-router-dom';
import { withAuthentication } from '../HOC';

class Navigation extends React.Component {
  handleLogout = async () => {
    await Auth.signOut();
    this.props.userHasAuthenticated(false);
    this.props.history.push('/login');
  }

  render() {
    return (
      <Navbar fluid collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <Link to="/">CodeNote</Link>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav pullRight>
            {this.props.isAuthenticated
              ? <NavItem onClick={this.handleLogout}>Logout</NavItem>
              : <Fragment>
                <LinkContainer to="/signup">
                  <NavItem>Signup</NavItem>
                </LinkContainer>
                <LinkContainer to="/login">
                  <NavItem>Login</NavItem>
                </LinkContainer>
              </Fragment>
            }
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    )
  }
}

export default withAuthentication(Navigation)
