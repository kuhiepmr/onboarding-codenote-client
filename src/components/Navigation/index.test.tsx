import { shallow } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import Navigation from './index';

describe('Navigation', () => {
  const mockStore = configureStore([])

  // first render without authentication
  it('should render success', () => {
    const initState = {
      authenticate: { isAuthenticated: false }
    }
    const store = mockStore(initState)
    const wrapper = shallow(<Navigation store={store} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('isAuthenticated')).toEqual(false)
  });

  // dispatch action authenticate
  it('should render success', () => {
    const initState = {
      authenticate: { isAuthenticated: true }
    }
    const store = mockStore(initState)
    const wrapper = shallow(<Navigation store={store} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('isAuthenticated')).toEqual(true)
  });
});