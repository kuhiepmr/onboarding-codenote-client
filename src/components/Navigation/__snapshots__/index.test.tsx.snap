// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render success 1`] = `
<Navigation
  isAuthenticated={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userHasAuthenticated={[Function]}
/>
`;

exports[`Navigation should render success 2`] = `
<Navigation
  isAuthenticated={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userHasAuthenticated={[Function]}
/>
`;
